test/test_jit.py

```diff
@@ -5261,6 +5261,48 @@ def foo(z):
             return y[0][1]
         self.checkScript(foo, ((1, [[1, 2], [3, 4]]),))
 
+    def test_nested_aug_assign(self):
+        @torch.jit.script
+        class SomeClass(object):
+            def __init__(self):
+                self.num = 99
+
+            def __iadd__(self, x):
```

Comment:
Can you add a test for a class which doesn't define `__iadd__`, but does define `__add__` ? 
```
class A(object):
    def __init__(self, num):
        self.a = num

    def __add__(self, other):
        return A(self.a + other.a)


class B(object):
    def __init__(self, my_a):
        self.my_a = my_a

    def do_plus_equal(self, a_val):
        self.my_a += a_val
        return self.my_a.a

b = B(A(4))
print(b.do_plus_equal(A(5)))
# 9
```